semgrep/semgrep/core_runner.py

```diff
@@ -298,14 +298,18 @@ def _parse_core_output(
     def _add_match_times(
         self,
         rule: Rule,
-        match_time_matrix: Dict[Tuple[str, str], float],
+        match_time_matrix: Dict[Tuple[str, str], Tuple[float, float, float]],
```

Comment:
Can we turn this type into an object/class `Dict[Tuple[str, str], Tuple[float, float, float]]`? It's difficult to understand what each of these pieces of data represent, and makes for very long type hints. It'll also be easier to extend in the future if we have to :+1:

Reply:
yep agree, we need to do the same also on the OCaml side. Those tuples are getting big.